contracts/token/ERC721/extensions/ERC721Royalty.sol

```diff
@@ -29,10 +29,12 @@ abstract contract ERC721Royalty is ERC2981, ERC721 {
     }
 
     /**
-     * @dev See {ERC721-_burn}. This override additionally clears the royalty information for the token.
+     * @dev See {ERC721-_update}. This override additionally clears the royalty information for the token after burning.
      */
-    function _burn(uint256 tokenId) internal virtual override {
-        super._burn(tokenId);
-        _resetTokenRoyalty(tokenId);
+    function _update(address from, address to, uint256 firstTokenId, uint256 batchSize) internal virtual override {
+        super._update(from, to, firstTokenId, batchSize);
+        if (to == address(0)) {
```

Comment:
if `to == address(0)` and `batchSize > 1`, we should probably do something. Revert?